lib/deactivation-survey/deactivation-survey.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ * Plguin Deactivation Survey Class.
+ */
+class Deactivation_Survey {
+
+    /**
+     * API Url.
+     *
+     * @var string
+     */
+    public $api_url = 'https://rtmedia-dev.rtm.rt.gw/wp-json/rtps/v1/survey'; // Replace the API Url by the production API Url.
+
+    /**
+     * Basic Auth Username
+     *
+     * @var string
+     */
+    private $auth_user = 'rabiul'; // Replace the admin user by the production admin user.
+
+    /**
+     * Basic Auth Password.
+     *
+     * @var string
+     */
+    private $auth_password = 'SxQf 1BT0 bYcZ YU3d cQkS aT3Z'; // Replcae the auth token by the production auth token.
```

Comment:
`$auth_user` and `$auth_password` should be taken as a user input in plugin settings page.
OR can be defined as a constant in `wp-config.php` and the constants can be used here.

Reply:
@elifvish 
These things will not provided to the client as it was required only for the API authentication & the user's of the plugin. So that users can be authenticated before posting any data.